lib/v0_bindings_extensions.rb

```diff
@@ -123,11 +133,21 @@ def update_model!(model)
 
   Api::V0::Bindings::GetHighlightsSummaryParameters.class_exec do
     def summarize(user_uuid:)
-      highlights = ::Highlight.by_user(user_uuid)
+      users = []
+
+      if sets.blank? or sets.include?('user:me')
+        users.push(user_uuid)
+      end
+      if  sets.to_a.include?('curated:openstax') and scope_id.present?
+        curator_scope = CuratorScope.find_by(:scope_id => scope_id)
+        users.push(curator_scope.curator_id) unless curator_scope.nil?
+      end
+
+      highlights = ::Highlight.by_user(users)
```

Comment:
Instead of a separate line here that kind of duplicates line 151 below, you could instead do

```ruby
filters = to_hash.except(:sets)
filters[:user] = sets_to_user_ids(sets) # or something
```

And then this line goes away and line 150 becomes

```ruby
filter.each do |key, value|
```